pkg/asset/kubeconfig/agent.go

```diff
@@ -0,0 +1,48 @@
+package kubeconfig
+
+import (
+	"fmt"
+	"strings"
+
+	agentmanifests "github.com/openshift/installer/pkg/agent/manifests"
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/tls"
+)
+
+// AgentAdminClient is the asset for the agent admin kubeconfig.
+type AgentAdminClient struct {
+	AdminClient
+}
+
+// Dependencies returns the dependency of the kubeconfig.
+func (k *AgentAdminClient) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&tls.AdminKubeConfigClientCertKey{},
+		&tls.KubeAPIServerCompleteCABundle{},
+	}
+}
+
+// Generate generates the kubeconfig.
+func (k *AgentAdminClient) Generate(parents asset.Parents) error {
+	ca := &tls.KubeAPIServerCompleteCABundle{}
+	clientCertKey := &tls.AdminKubeConfigClientCertKey{}
+	parents.Get(ca, clientCertKey)
+
+	// TODO(zaneb): Get this from a ClusterDeployment Asset instead
+	clusterDeployment, err := agentmanifests.GetClusterDeployment()
+	if err != nil {
+		return err
+	}
+
+	clusterName := clusterDeployment.Spec.ClusterName
+	extAPIServerURL := fmt.Sprintf("https://api.%s.%s:6443", clusterName, strings.TrimSuffix(clusterDeployment.Spec.BaseDomain, "."))
+
+	return k.kubeconfig.generate(
+		ca,
+		clientCertKey,
+		extAPIServerURL,
+		clusterName,
+		"admin",
+		kubeconfigAdminPath,
+	)
+}
```

Comment:
It looks like there isn't a Name() or Load() function for this asset, would it be possible to add for consistency.

Reply:
It's inherited from the regular `AdminClient`.